rpc-server/src/modules/blocks/utils.rs

```diff
@@ -6,6 +6,47 @@ use anyhow::Context;
 use near_primitives::views::{StateChangeValueView, StateChangesRequestView};
 use num_traits::ToPrimitive;
 
+async fn fetch_object_from_s3(
+    s3_client: &aws_sdk_s3::Client,
+    s3_bucket_name: &str,
+    key: String,
+) -> anyhow::Result<aws_sdk_s3::output::GetObjectOutput> {
+    let mut s3_response = s3_client
```

Comment:
This doesn't seem idiomatic to me.

Can you consider refactoring it like the following:

```rust
let mut retry_count: u8 = 0;

let s3_response = loop {
    match s3_client
        .get_object()
        .bucket(s3_bucket_name)
        .key(&key)
        .send()
        .await
    {
        Ok(response) => break Ok(response),
        Err(err) => match err {
            // TODO: add a comment explaining why retry on these errors
            aws_sdk_s3::types::SdkError::ConstructionFailure(_)
            | aws_sdk_s3::types::SdkError::TimeoutError(_)
            | aws_sdk_s3::types::SdkError::DispatchFailure(_)
            | aws_sdk_s3::types::SdkError::ResponseError(_) => {
                tracing::debug!("Error to get object from s3, Retry");
                if retry_count < 3 {
                    retry_count += 1;
                    continue;
                } else {
                    break Err(err);
                }
            }
            // TODO: add a comment why don't we retry on other errors
            _ => break Err(err),
        },
    }
};
```

Also, since we're passing the `key` already, we expect to know what kind of object we're fetching when we call the function. How about adding a generic to the function so we can pass the `*View` type for the fetched object to be parsed to?

For instance:

```rust
let block = fetch_object_from_s3::<BlockView>(&client, &bucket_name, format!("{:0>12}/block.json", block_height))?;
```